tests/i9276.scala

```diff
@@ -0,0 +1,13 @@
+import scala.deriving.*
+
+sealed trait A
+
+case class C() extends A
+
+sealed trait B extends A
+
+case class D() extends B
+
+@main def Test =
+  summon[Mirror.Of[A]]
+  summon[Mirror.Of[B]]
```

Comment:
Can we maybe add another test like the following?

~~~ scala
val mirrorA = summon[Mirror.Of[A]]
summon[mirrorA.MirroredElemTypes <:< (C *: B *: EmptyTuple)]
~~~